conda/core/linked_data.py

```diff
@@ -105,6 +105,18 @@ def all_subdir_urls(self):
                 subdir_urls.add(subdir_url)
         return subdir_urls
 
+    def query(self, package_ref_or_match_spec):
+        # returns a generator
+        param = package_ref_or_match_spec
+        if isinstance(param, string_types):
+            param = MatchSpec(param)
+        if isinstance(param, MatchSpec):
+            return (prefix_rec for prefix_rec in self.iter_records()
+                    if param.match(prefix_rec))
+        else:
+            # assume isinstance(param, PackageRef)
```

Comment:
`# assume` => `assert`?

Reply:
Sure, yeah.  And in the other two `query()` API methods as well.